Core/Tests/Speckle.Core.Tests.Unit/Helpers/BinaryFormatter.cs

```diff
@@ -0,0 +1,100 @@
+using System.Security.Cryptography;
+using NUnit.Framework;
+using Speckle.Core.Helpers;
+
+namespace Speckle.Core.Tests.Unit.Helpers;
+
+[TestFixture]
+[TestOf(nameof(BinaryFormatter))]
+public class BinaryFormatterTests
+{
+  [Test]
+  public void ZeroLengthStringParityCheck() => CompareTechniques(new string('b', 0));
+
+  /// <summary>
+  /// Refer to https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-nrbf/10b218f5-9b2b-4947-b4b7-07725a2c8127 for the specifics of this requirement.
+  /// </summary>
+  [Test]
+  public void SingleByteLengthStorageParityCheck() => CompareTechniques(new string('b', 127));
+
+  /// <summary>
+  /// Refer to https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-nrbf/10b218f5-9b2b-4947-b4b7-07725a2c8127 for the specifics of this requirement.
+  /// </summary>
+  [Test]
+  public void TwoByteLengthStorageParityCheck() => CompareTechniques(new string('b', 128));
+
+  /// <summary>
+  /// Refer to https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-nrbf/10b218f5-9b2b-4947-b4b7-07725a2c8127 for the specifics of this requirement.
+  /// </summary>
+  [Test]
+  public void ThreeByteLengthStorageParityCheck() => CompareTechniques(new string('b', 16384));
+
+  /// <summary>
+  /// Refer to https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-nrbf/10b218f5-9b2b-4947-b4b7-07725a2c8127 for the specifics of this requirement.
+  /// </summary>
+  [Test]
+  public void FourByteLengthStorageParityCheck() => CompareTechniques(new string('b', 2097152));
+
+  /// <summary>
+  /// Refer to https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-nrbf/10b218f5-9b2b-4947-b4b7-07725a2c8127 for the specifics of this requirement.
+  /// </summary>
+  [Test]
+  public void FiveByteLengthStorageParityCheck() => CompareTechniques(new string('b', 268435456));
```

Comment:
My only concern with this test is the memory profile on your testing infrastructure, but also, what do I know about these things?